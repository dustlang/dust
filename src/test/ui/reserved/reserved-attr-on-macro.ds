#[dustc_attribute_should_be_reserved]
//~^ ERROR cannot find attribute `dustc_attribute_should_be_reserved` in this scope
//~| ERROR attributes starting with `dustc` are reserved for use by the `dustc` compiler

macro_rules! foo {
    () => (());
}

fn main() {
    foo!(); //~ ERROR cannot determine resolution for the macro `foo`
}
