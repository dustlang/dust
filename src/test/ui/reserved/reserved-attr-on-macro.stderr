error: attributes starting with `dustc` are reserved for use by the `dustc` compiler
  --> $DIR/reserved-attr-on-macro.rs:1:3
   |
LL | #[dustc_attribute_should_be_reserved]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot determine resolution for the macro `foo`
  --> $DIR/reserved-attr-on-macro.rs:10:5
   |
LL |     foo!();
   |     ^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `dustc_attribute_should_be_reserved` in this scope
  --> $DIR/reserved-attr-on-macro.rs:1:3
   |
LL | #[dustc_attribute_should_be_reserved]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

